install/kubernetes/cilium/values.yaml

```diff
@@ -29,7 +29,7 @@ cluster:
   name: default
   # -- (int) Unique ID of the cluster. Must be unique across all connected
   # clusters and in the range of 1 to 255. Only required for Cluster Mesh.
-  id:
+  id: 0
```

Comment:
What occurs when this is empty? It doesn't get the zero value of "0" when parsed into Go structs?

Reply:
When using the ConfigMap, the `cluster-id` option is loaded from the relevant file. Without the `0`, the file is empty, so Viper (silently) fails to convert an empty string to an int, here:

https://github.com/cilium/cilium/blob/c3cd7da63885a320435a63a70cd7d9c7c5971abd/pkg/option/config.go#L2764

Prior to this PR, nothing too bad was happening: the default value for `cluster-id` is `0` and this is also the Go zero value for the `int` type. Thus, viper was failing silently, but the `c.ClusterID` was left with a `0` anyway.

But the proposed changes strengthen the validation process on option values. Without specifying a correct value, we would end up with this error:

`option cluster-id: unable to cast "" of type string to int64`

Reply:
Okay I see, and I agree with your reasoning. LGTM.